src/main/scala/com/scalapenos/riak/internal/RiakHttpClientHelper.scala

```diff
@@ -165,7 +164,9 @@ private[riak] class RiakHttpClientHelper(system: ActorSystem) extends RiakUriSup
 
     val entity = JsObject("props" -> JsObject(newProperties.map(property ⇒ (property.name -> property.json)).toMap))
 
-    httpRequest(Put(PropertiesUri(server, bucket), entity)).map { response ⇒
+    // For some reason, Riak set bucket props HTTP endpoint doesn't handle compressed request properly.
+    // So we disable compression for this request unconditionally.
```

Comment:
Please create an issue for this and reference it here so we can keep track of it. Perhaps also create an issue for Riak itself and link the two?